# Derived from conda-forge/sqlean.py-feedstock
# Original file: recipe/0002-fix-constans-in-sqlean-time-c.patch
# License: BSD-3-Clause
# See: third_party\conda-forge\LICENSE
#
--- a/sqlean-src/src/time/duration.c	2026-06-12 16:08:09.475965200 +0900
+++ b/sqlean-src/src/time/duration.c	2026-06-12 16:09:35.804083200 +0900
@@ -12,12 +12,21 @@
 #include "time/timex.h"
 
 // Common durations.
+#if defined(_MSC_VER)
+#define Nanosecond   ((Duration)1)
+#define Microsecond  ((Duration)1000)
+#define Millisecond  ((Duration)1000000)
+#define Second       ((Duration)1000000000)
+#define Minute       ((Duration)(60) * Second)
+#define Hour         ((Duration)(60) * Minute)
+#else
 const Duration Nanosecond = 1;
 const Duration Microsecond = 1000 * Nanosecond;
 const Duration Millisecond = 1000 * Microsecond;
 const Duration Second = 1000 * Millisecond;
 const Duration Minute = 60 * Second;
 const Duration Hour = 60 * Minute;
+#endif
 
 #pragma region Conversion
 
